### ISSUE TITLE
[sign_in_with_apple] fullName doesnt work 

### ISSUE THREAD

**Issue by cyberpwnn** [aboutyou/dart_packages#354]
Real devices, production,debug, doesnt matter. It just doesnt work.

Reopen #250. Not sure why it was closed, doesnt work.

**Comment by ahmadhassan7** [comment]
+1 its not working for me too

**Comment by MendleM** [comment]
Did you find a workaround?

**Comment by kyadalu1** [comment]
Try this https://www.youtube.com/watch?v=v_Bu2jes3YE

**Comment by Goodstuff4UonYT** [comment]
Any updates? Apple rejects the app update because of this and we don't get one released anymore...

**Comment by Goodstuff4UonYT** [comment]
I have just created a new Apple ID and signed in with the new Apple ID. I was also asked for the name when I logged in and fullname is still null.

**Comment by adminptl0001** [comment]
> I have just created a new Apple ID and signed in with the new Apple ID. I was also asked for the name when I logged in and fullname is still null.

I don't think this is correct. I was in your exact position with App Store review. AuthorizationCredentialAppleID has givenName and familyName upon first sign in after removing the account from Apple Login. 

My issue was that Firebase Authentication was not reading those values into displayName as it does with Google Sign In. I solved this by caching the values from apple with unique id the first time they are provided and referencing them later on when needed.

**Comment by avdept** [comment]
Ok, the problem I had - I originally didn't add proper scopes and my iCloud account was logged in without name. It was all the same until I removed login via my settings in iCloud account. Next time it asked for name and email correctly.

**Comment by memishood** [comment]
I have the same issue, my server can read the email, but cannot read the fullName when using this package. It is always null even though I've added fullName to scopes.

**Comment by equescodebelike** [comment]
Ensure that your app relays the credentials and user information to your app servers. The API collects this information and shares it with your app the first time the user logs in using Sign in with Apple. If the user then uses Sign in with Apple on another device, the API doesn’t ask for the user’s name or email again. It collects the information again only if the user stops using Sign in with Apple and later reconnects to your app.

https://developer.apple.com/documentation/sign_in_with_apple/authenticating-users-with-sign-in-with-apple#Send-information-to-app-servers-and-verify-tokens

**Comment by geekswamp** [comment]
Any update? I got the same issue. The `givenName` and `familyName` from the `AuthorizationCredentialAppleID` instance are null.

```dart
final String rawNonce = nonce();
final AuthorizationCredentialAppleID credential =
   await SignInWithApple.getAppleIDCredential(
       nonce: sha256String(rawNonce),
       scopes: [
            AppleIDAuthorizationScopes.fullName,
            AppleIDAuthorizationScopes.email,
        ],
   );

final AuthCredential authCredential =
    AppleAuthProvider.credentialWithIDToken(
       credential.identityToken!,
       rawNonce,
       AppleFullPersonName(
            givenName: credential.givenName,
            familyName: credential.familyName,
        ),
    );
```
